pkg/frontend/clustermanager_get.go

```diff
@@ -20,19 +21,100 @@ func (f *frontend) getClusterManagerConfiguration(w http.ResponseWriter, r *http
 	ctx := r.Context()
 	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
 	vars := mux.Vars(r)
+	badRequestError := fmt.Sprintf("The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["ocmResourceType"], vars["resourceProviderNamespace"], vars["api-version"])
+	unsupportedRequestError := fmt.Sprintf("The resource type '%s' is not supported for api version '%s'.", vars["ocmResourceType"], vars["api-version"])
 
-	if f.apis[vars["api-version"]].ClusterManagerConfigurationConverter == nil {
-		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
+	var (
+		b   []byte
+		err error
+	)
+
+	switch vars["ocmResourceType"] {
+	case "syncset":
```

Comment:
can we encapsulate this logic of valid resources / non nil converts into a shared function to reduce the duplication between our REST operations? It might also help down the line if we want to support new resource types